dynawo/sources/Solvers/Common/DYNSolverImpl.cpp

```diff
@@ -261,13 +261,16 @@ Solver::Impl::evalZMode(vector<state_g> &G0, vector<state_g> &G1, const double &
     model_->evalZ(time);
     ++stats_.nze_;
 
-    if (model_->getSilentZChange())
-      state_.setFlags(SilentZChange);
+    if (model_->getSilentZChange(NotUsedInDiscreteEquations))
+      state_.setFlags(SilentZNotUsedInDiscreteEqChange);
 
     if (model_->zChange()) {
       model_->evalG(time, G1);
       ++stats_.nge_;
-      state_.setFlags(ZChange);
+      if (model_->getSilentZChange(NotUsedInContinuousEquations))
+        state_.setFlags(SilentZNotUsedInContinuousEqChange);
+      else
+        state_.setFlags(ZChange);
       nonSilentZChange = true;
```

Comment:
I would suggest a simplest implementation (if it works):
if model_->getSilentZChange(NonSilentZ) {
  updateG(); (everything that has to be done)
  state_->setFlags()
} else if (model_->getSilentZChange(NotUsedInContinuousEquations) {
  updateG();
  state->setFlags();
} else if {model_->getSilentZChange(NotUsedInDiscreteEquations) {
  state_->setFlags();
}

Reply:
NotUsedInDiscreteEquations should be set independently from the others, so this would not work

Reply:
As of today, with the algorithm in ModelMulti, notUsedInDiscreteEquations can't be set if any of the two others is set (if I understand it right)

My main point is that I personnally think that dividing the logic and testing makes it more complicated to understand the whole behavior: depending on the z change status, we should do different actions and the idea is that this should be clear at first glance.

Reply:
Here are the rules:
notUsedInDiscreteEquations can have any value at any time
SilentZNotUsedInContinuousEqChange and NotSilentZChange cannot be true at the same time

Reply:
done